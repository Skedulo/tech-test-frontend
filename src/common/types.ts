export interface IAppTabContainer {
  service: IDataService
}

export interface IDataService {}
