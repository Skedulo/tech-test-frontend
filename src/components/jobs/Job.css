.jobs {
  padding: 0;
  margin: 0;
  list-style: none;
}

.job {
  padding: 12px;
  margin-bottom: 8px;
  background-color: #ffffff;
}

.job__info {
  margin-bottom: 8px;
}

.job__info__name {
  font-size: 16px;
}

.job__info__id {
  font-size: 16px;
  color: #8d8383;
}

.job__info__contact {
  font-size: 14px;
  color: #8d8383;
}

.job__description {
  display: flex;
}

.job__info__date {
  flex: 1;
  font-size: 14px;
  color: #8d8383;
}

.job__resources {
  flex: 1;
  text-align: right;
}

.job__resource {
  display: inline-flex;
  align-items: center;
  justify-content: center;
  width: 30px;
  height: 30px;
  border-radius: 50%;
  border: 1px solid #3d9bff;
  color: #3d9bff;
  font-weight: bold;
  font-size: 14px;
}
