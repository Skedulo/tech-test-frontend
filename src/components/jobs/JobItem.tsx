import React from "react"
import { Contact, Job } from "../../common/types";
import moment from 'moment';

export interface JobItemProps {
    item: Pick<Job, 'name' | 'start' | 'end'> & { contact: Contact }
}

export function JobItem({ item }: JobItemProps) {
    const { name, contact, start, end } = item;

    const dateTimeTemplate = () => {
        const startTime = moment(start);
        const endTime = moment(end);
        return (
            <div className="job__job-time">
                <div data-testid="datetime-from" className="job__job-time__value">
                    <div className="bold-600">From</div>
                    <div>{startTime.format('ddd MMM DD YYYY hh:mm A')}</div>
                </div>
                <div data-testid="datetime-to" className="job__job-time__value">
                    <div className="bold-600">To</div>
                    <div>{endTime.format('ddd MMM DD YYYY hh:mm A')}</div>
                </div>
            </div>
        )
    }

    return (
        <div className='job flex-columns-2'>
            <div data-testid='job-name' className="text-x25">
                {name}
            </div>
            <br />
            {dateTimeTemplate()}
            <br />
            <div data-testid='contact'><span className="bold-600">Contact: </span>{contact.name}</div>
        </div>
    );
}