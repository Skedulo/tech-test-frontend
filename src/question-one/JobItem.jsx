import React from 'react'

export default ({item}) => (
  <div>
    <div>Jobs: {item.name}</div>
    <div>Start: {item.start}</div>
    <div>End: {item.end}</div>
    <div>Contact: {item.contact && item.contact.name}</div>
  </div>
)