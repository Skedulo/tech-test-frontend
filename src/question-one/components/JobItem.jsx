import React from 'react'
import PropTypes from 'prop-types'

const JobItem = ({ item }) => (
  <div>
    <div>Jobs: {item.name}</div>
    <div>Start: {item.start}</div>
    <div>End: {item.end}</div>
    <div>Contact: {item.contact && item.contact.name}</div>
  </div>
)

JobItem.propTypes = {
  item: PropTypes.shape({
    name: PropTypes.string,
    start: PropTypes.oneOfType([
      PropTypes.string,
      PropTypes.instanceOf(Date)
    ]),
    end: PropTypes.oneOfType([
      PropTypes.string,
      PropTypes.instanceOf(Date)
    ]),
    contact: PropTypes.shape({
      name: PropTypes.string
    })
  })
}

export default JobItem
