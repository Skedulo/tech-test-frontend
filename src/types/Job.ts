import {JobAllocation} from './JobAllocation'
import {Contact} from './Contact'

export interface Job {
    id?: string
    name?: string
    contact?: Contact
    location?: string
    jobAllocations?: [JobAllocation]
    start: string
    end: string
}