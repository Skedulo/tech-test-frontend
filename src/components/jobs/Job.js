import React from 'react';

import { formatDate } from '../../utils';

import './Job.css';

export const Jobs = ({ className = '', jobs }) => (
  <ul className={`jobs ${className}`}>
    {jobs.map(job => (
      <Job key={job.id} job={job}></Job>
    ))}
  </ul>
);

export const Job = ({ job }) => {
  const startDate =
    formatDate(job.start).getDayName() +
    ' ' +
    formatDate(job.start).getFullMonth() +
    ' ' +
    formatDate(job.start).getDate() +
    ' ' +
    formatDate(job.start).getFullYear();

  const timeRange = `
    ${formatDate(job.start).getUTCTime()} - ${formatDate(job.end).getUTCTime()}
  `;
  return (
    <li key={job.id} className="job">
      <div className="job__info">
        <div>
          <strong className="job__info__name">{job.name}</strong>
          <span className="job__info__id"> (Job #{job.id})</span>
        </div>
        <span className="job__info__contact">{job.contact.name}</span>
      </div>
      <div className="job__description">
        <div className="job__info__date">
          <div>
            <span>{startDate}</span>
          </div>
          <strong>{timeRange}</strong>
        </div>
      </div>
    </li>
  );
};
