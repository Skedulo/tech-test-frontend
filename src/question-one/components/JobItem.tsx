import React from 'react'
import { Job } from '../../types/Job'

interface JobItemProps {
  item: Job
}

const JobItem : React.FC<JobItemProps>= ({ item }) => (
  <div>
    <div>Jobs: {item.name}</div>
    <div>Start: {item.start}</div>
    <div>End: {item.end}</div>
    <div>Contact: {item.contact && item.contact.name}</div>
  </div>
)

export default JobItem
