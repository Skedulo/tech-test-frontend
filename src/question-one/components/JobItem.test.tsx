import React from 'react'
import {create} from 'react-test-renderer'

import JobItem from './JobItem'
import { Job } from '../../types/Job'

describe('JobItem', () => {
    it('should render correctly', () => {
        const item = {
            name: 'job',
            start: 'start',
            end: 'end',
            contact: {
                name : 'contactName'
            },
        } as Job
        const jobItem = create(<JobItem item={item}/>)
        
        expect(jobItem.toJSON()).toMatchSnapshot()
    })
})