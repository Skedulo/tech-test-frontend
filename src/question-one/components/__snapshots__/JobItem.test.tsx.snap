// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JobItem should render correctly 1`] = `
<div
  title="Job Item: job"
>
  <div>
    Jobs: 
    job
  </div>
  <div>
    Start: 
    start
  </div>
  <div>
    End: 
    end
  </div>
  <div>
    Contact: 
    contactName
  </div>
</div>
`;
